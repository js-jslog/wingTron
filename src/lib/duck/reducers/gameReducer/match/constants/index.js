// @flow

import type { Match } from '../types'

export const EXAMPLE_MATCH = {
  field_width: 200,
  field_height: 200,
  matches: 10
}

