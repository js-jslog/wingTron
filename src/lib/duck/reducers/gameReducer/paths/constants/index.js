// @flow

import type { Coord } from '../types'
import type { Path } from '../types'
import type { NamedPath } from '../types'
import type { PathsById } from '../types'
import type { PathList } from '../types'
import type { Paths } from '../types'

const EXAMPLE_COORD: Coord = [ 150, 100 ]
const EXAMPLE_PATH: Path = [
  EXAMPLE_COORD,
  EXAMPLE_COORD
]
const EXAMPLE_NAMED_PATH0: NamedPath = {
  id: '0',
  path: EXAMPLE_PATH
}
const EXAMPLE_NAMED_PATH1: NamedPath = {
  id: '1',
  path: EXAMPLE_PATH
}
const EXAMPLE_PATHS_BY_ID: PathsById = {
  '0': EXAMPLE_NAMED_PATH0,
  '1': EXAMPLE_NAMED_PATH1
}
const EXAMPLE_PATH_LIST: PathList = [ '0', '1' ]
export const EXAMPLE_PATHS = {
  byId: EXAMPLE_PATHS_BY_ID,
  allIds: EXAMPLE_PATH_LIST
}
