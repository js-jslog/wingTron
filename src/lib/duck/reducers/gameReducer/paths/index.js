// @flow

import type { Paths } from './types'
import type { Action } from '~/common/flow-types'
import { coordsToInitialPath } from './coordsToInitialPath'
import { pathArrayToPaths } from './pathArrayToPaths'

const getCoords = (player_options) => ([
  player_options.start_coord_x,
  player_options.start_coord_y
])

export const paths = (state_in: ?Paths=null, action: Action): ?Paths => {
  switch(action.type) {

    case 'START_GAME_FROM_OPTIONS': {

      const { players } = action.options
      const path_array = players.map(player_options => coordsToInitialPath(getCoords(player_options)))
      const paths = pathArrayToPaths(path_array)

      return paths
    }

    default: return state_in
  }
}
