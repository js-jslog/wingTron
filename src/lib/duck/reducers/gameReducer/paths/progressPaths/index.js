// @flow

import type { Paths } from '../types'
import type { Players } from '~/common/flow-types'
import { getRelevantPlayerDirection } from './getRelevantPlayerDirection'
import { progressPathInDirection } from './progressPathInDirection'

export const progressPaths = (players: Players, paths: Paths): Paths => {

  const paths_out = { ...paths }

  paths_out.allIds.forEach(path_id => {
    const direction = getRelevantPlayerDirection(players, path_id)
    // TODO; need to handle the thrown error from getRelevantPlayerDirection
    // so that direction is known to be defined
    // $FlowFixMe
    const path = progressPathInDirection(paths.byId[path_id].path, direction)
    paths_out.byId[path_id] = {
      id: path_id,
      path
    }
  })

  return paths_out
}

