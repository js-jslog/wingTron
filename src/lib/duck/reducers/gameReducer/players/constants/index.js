// @flow

import type { Player } from '../types'
import type { PlayersById } from '../types'
import type { PlayerList } from '../types'
import type { Players } from '../types'

const EXAMPLE_PLAYER0: Player = {
  id: '0',
  path: '0',
  direction: 0,
  turn_left_keycode: 37,
  turn_right_keycode: 39,
  colour: 'rgba(255,0,0, 0.5)',
}

const EXAMPLE_PLAYER1: Player = {
  id: '1',
  path: '1',
  direction: Math.PI,
  turn_left_keycode: 65,
  turn_right_keycode: 68,
  colour: 'rgba(0,0,255, 0.5)',
}

const EXAMPLE_PLAYERS_BY_ID: PlayersById = {
  '0': EXAMPLE_PLAYER0,
  '1': EXAMPLE_PLAYER1
}
const EXAMPLE_PLAYER_LIST: PlayerList = [ '0', '1' ]

export const EXAMPLE_PLAYERS: Players = {
  byId: EXAMPLE_PLAYERS_BY_ID,
  allIds: EXAMPLE_PLAYER_LIST
}
