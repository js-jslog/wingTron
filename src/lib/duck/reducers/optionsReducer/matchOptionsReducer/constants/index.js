// @flow

import type { MatchOptions } from '../types'

export const INITIAL_STATE: MatchOptions = {
  field_width: '200',
  field_height: '200',
  matches: '10'
}
