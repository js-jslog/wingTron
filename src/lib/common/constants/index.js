// @flow

import type { GameMatch, Player } from '~/common/flow-types'

import { INITIAL_STATE as EXAMPLE_MATCH_OPTIONS_IMPORT } from '~/duck/reducers/optionsReducer/matchOptionsReducer/constants'
import { INITIAL_PLAYER1_OPTIONS as EXAMPLE_PLAYER1_OPTIONS_IMPORT } from '~/duck/reducers/optionsReducer/playerOptionsReducer/constants'
import { INITIAL_PLAYER2_OPTIONS as EXAMPLE_PLAYER2_OPTIONS_IMPORT } from '~/duck/reducers/optionsReducer/playerOptionsReducer/constants'
import { EXAMPLE_PLAYERS as EXAMPLE_PLAYERS_IMPORT } from '~/duck/reducers/gameReducer/players/constants'

export const EXAMPLE_MATCH_OPTIONS = EXAMPLE_MATCH_OPTIONS_IMPORT
export const EXAMPLE_PLAYER1_OPTIONS = EXAMPLE_PLAYER1_OPTIONS_IMPORT
export const EXAMPLE_PLAYER2_OPTIONS = EXAMPLE_PLAYER2_OPTIONS_IMPORT

export const EXAMPLE_OPTIONS = {
  match: { ...EXAMPLE_MATCH_OPTIONS },
  players: [
    { ...EXAMPLE_PLAYER1_OPTIONS },
    { ...EXAMPLE_PLAYER2_OPTIONS }
  ]
}

export const EXAMPLE_GAME_MATCH: GameMatch = {
  field_width: 200,
  field_height: 200,
  matches: 10
}

export const EXAMPLE_PLAYERS = EXAMPLE_PLAYERS_IMPORT

export const EXAMPLE_GAME = {
  match: { ...EXAMPLE_GAME_MATCH },
  players: EXAMPLE_PLAYERS
}

