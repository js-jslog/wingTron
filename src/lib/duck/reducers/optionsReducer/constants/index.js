// @flow

import type { Options } from '../'
import { INITIAL_STATE as EXAMPLE_MATCH_OPTIONS } from '../matchOptionsReducer/constants'
import { INITIAL_STATE as EXAMPLE_PLAYER_OPTIONS } from '../playerOptionsReducer/constants'

export const EXAMPLE_OPTIONS: Options = {
  match: EXAMPLE_MATCH_OPTIONS,
  players: EXAMPLE_PLAYER_OPTIONS
}
