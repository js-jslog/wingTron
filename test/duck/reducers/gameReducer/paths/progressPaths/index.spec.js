// @flow

import { EXAMPLE_PLAYERS } from '~/common/constants'
import { EXAMPLE_PATHS } from '~/common/constants'
import { progressPaths } from '~/duck/reducers/gameReducer/paths/progressPaths'

describe('the function to progress the Paths object in the state', () => {

  it('should progress some basic paths as it has done previously', () => {

    const players = EXAMPLE_PLAYERS
    const paths = EXAMPLE_PATHS

    const progressed_paths = progressPaths(players, paths)

    expect(progressed_paths).toMatchSnapshot()
  })
})
