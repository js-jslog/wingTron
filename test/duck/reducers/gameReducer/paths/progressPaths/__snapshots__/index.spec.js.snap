// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the function to progress the Paths object in the state should progress some basic paths as it has done previously 1`] = `
Object {
  "allIds": Array [
    "0",
    "1",
  ],
  "byId": Object {
    "0": Object {
      "id": "0",
      "path": Array [
        Array [
          152,
          100,
        ],
        Array [
          151,
          100,
        ],
      ],
    },
    "1": Object {
      "id": "1",
      "path": Array [
        Array [
          148,
          100,
        ],
        Array [
          149,
          100,
        ],
      ],
    },
  },
}
`;
